File: ts/src/handlers/http/set_cors.test.ts
import { FromCommandLine } from "../../app.js";
import { MockApiHandler, MockHttpRequest } from "../../testutils.js";
import { SetCorsHeader } from "./set_cors.js";

describe("TestSetCors", () => {
  it("SetCors", async () => {
    const hdl = new MockApiHandler<never, never>(FromCommandLine([]), MockHttpRequest());
    expect(await SetCorsHeader(hdl)).toBeTruthy();
    expect(hdl.Response().Header().Get("Access-Control-Allow-Origin")).toBe("*");
  });
});

File: go/handlers/http/set_cors_test.go
package handlers_http

import (
	"net/http"
	"testing"

	"github.com/mabels/runified/testutils"
)

func TestSetCords(t *testing.T) {
	api := &testutils.MockApi{}
	r, err := http.NewRequest("OPTIONS", "test://test", nil)
	if err != nil {
		t.Fatal(err)
	}
	w := testutils.NewMockResponseWriter()
	hdl := &testutils.MockApiHandler[struct{}, struct{}]{
		ApiRef: api,
		Req:    r,
		Res:    w,
	}
	SetCorsHeader(hdl)

	if w.Header().Get("Access-Control-Allow-Origin") != "*" {
		t.Error("Access-Control-Allow-Origin not set")
	}

}
